### PR TITLE
Action var tests off

### PR DESCRIPTION
**Pull request by BrianCashProf**
## What does this implement/fix?

_Put a `x` in the boxes that apply_

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds a feature)
- [ ] Breaking change (fix or feature that would cause existing features not to work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Other (please describe):

## Testing

- [x] Did you test the pull request locally?
- [ ] Did you add new tests?

## Any other comments?

Following the changes introduced in https://github.com/nebari-dev/nebari/pull/2569, We have decided to use action variables instead of local environment variables. 
These action variables must be set at the repo level and can be set by forks differently to the main repository without merge conflicts. 
The action variable is set as a binary boolean (with 0 being false and 1 being true).

### CODE REVIEW

**Comment by viniciusdc** [comment]
Hi @BrianCashProf, thanks for opening the PR; I wonder if the variable could be boolean instead of the int. Also, while reading the [docs](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/store-information-in-variables#using-the-vars-context-to-access-configuration-variable-values) for variables again, I noticed this:
> If a configuration variable has not been set, the return value of a context referencing the variable will be an empty string.

So, one thing that we could also change is the primary condition, so instead of looking for a value of the given variable (and this requiring that to be preset), we could use:
```
if: ${{ ! vars.NO_PROVIDER_CREDENTIALS_*** }}
```

**Comment by BrianCashProf** [comment]
> Hi @BrianCashProf, thanks for opening the PR; I wonder if the variable could be boolean instead of the int. Also, while reading the [docs](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/store-information-in-variables#using-the-vars-context-to-access-configuration-variable-values) for variables again, I noticed this:
> 
> > If a configuration variable has not been set, the return value of a context referencing the variable will be an empty string.
> 
> So, one thing that we could also change is the primary condition, so instead of looking for a value of the given variable (and this requiring that to be preset), we could use:
> 
> ```
> if: ${{ ! vars.NO_PROVIDER_CREDENTIALS_*** }}
> ```
> 
> I've attested that this works as expected, but with the caveat that those variables must be set in the repository settings of a given fork.

I'll take a look into this method.

**Comment by tylergraff** [comment]
> If by default we want to run the pipeline and the flag only needs to be placed to skip the pipeline; then this is my suggestion

@BrianCashProf Yes we do want to run by default and only skip if the flag is set accordingly. Please make your suggested change. Once that is committed, is there anything else needed before this PR can be merged?

**Review by viniciusdc — COMMENTED** [review]
@BrianCashProf, make sure they follow the standard you suggested above. `NO_PROVIDER_CREDENTIALS_AWS != 'true'` to `vars.NO_PROVIDER_CREDENTIALS_AZURE == 0`. Also, now that we removed the previous `check-for-credentials` job and we are also relying on different variables for each provider anyway, I suggest that we rename the variable to `SKIP_***_INTEGRATION_TEST`.

File: .github/workflows/test_aws_integration.yaml [hunk 1]
```diff
@@ -33,31 +33,12 @@ env:
...
-      - name: Check if user wants to run AWS integration based on credentials
-        id: flag-check
-        run: |
-          if [ "${{ env.NO_PROVIDER_CREDENTIALS_aws }}" == "true" ]; then
-            echo "::set-output name=provider_credentials_aws::0"
-          else
-            echo "::set-output name=provider_credentials_aws::1"
-          fi
-
+# Must have action variable NO_PROVIDER_CREDENTIALS_AWS set up in the repo as a binary boolean.
```
Comment (viniciusdc) [inline .github/workflows/test_aws_integration.yaml:38]: Is this still required?

Comment (BrianCashProf) [reply .github/workflows/test_aws_integration.yaml]: Nope, I've removed the comment.

**Review by viniciusdc — APPROVED** [review]
LGTM, thanks @BrianCashProf

**Pull request merged**